### DISCUSSION TITLE
Default value for `Worker.join` timeout

### DISCUSSION THREAD

**Discussion by bicarlsen** [pymeasure/pymeasure#704] [General]
Currently the default value for [`Worker.join`](https://github.com/pymeasure/pymeasure/blob/4770fe54be14b2c1618df9711b197330dbd3f262/pymeasure/experiment/workers.py#L98) is set to `0` meaning the [procedure should stop immediately](https://docs.python.org/3/library/threading.html#threading.Thread.join). It seems like this could be confusing as I, and I think most people, assume that not specifying a timeout would allow the worker to run indefinitely. This behavior is also inherited from [`StoppableThread`](https://github.com/pymeasure/pymeasure/blob/4770fe54be14b2c1618df9711b197330dbd3f262/pymeasure/thread.py#L61).

To me it would make more sense to set these default values to `None`, although this could be a major breaking change.

**Comment by msmttchr** [comment]
See also #273, where this topic has been discussed.
